src/Core/Resettable.php

```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace SilverStripe\Core;
+
+/**
+ * Represents a class with a local cache which normally invalidates itself between requests.
+ *
+ * Designed so that tests can automatically flush these objects between runs in lieu
+ * of a real change.
+ */
+interface Resettable
```

Comment:
Shouldn't we move this stuff to explicit registration via YAML config? My understanding was that `ClassInfo::implementorsOf()` is on it's way out, because it forces us to parse all PHP files on a flush (which is no longer required for autoloading thanks to PSR-7). The `interface Resettable` is fine, but I wouldn't use `ClassInfo::implementorsOf('Resettable')` to retrieve those classes.